### DISCUSSION TITLE
Request for Multi-Monitor Support and touch driver like SuperDisplay app.

### DISCUSSION THREAD

**Discussion by utkvishwas** [LizardByte/community#205] [Sunshine]
I am writing to request a new feature for Sunshine - multi-monitor support. As a user, I often find myself working with multiple displays, and it would be extremely helpful if I could extend the Sunshine display to cover all of my screens.

Currently, the Sunshine and Moonlight display is limited to a mirror screen, which makes it difficult to work on multiple tasks simultaneously. With multi-monitor support, I could easily drag Sunshine across all of my displays and have a larger workspace to work with. This would greatly enhance my productivity and make Sunshine and Moonlight even more valuable to me.

I understand that implementing multi-monitor support may require significant effort on your part, but I believe it would be a worthwhile investment. Many users would benefit from this feature. 

I know moonlight have to work on this too but you guys work together now I believe and hope it will not be that difficult.
Thank you for considering my request. I look forward to hearing your thoughts on this matter.

**Comment by LachlanStone** [comment]
Agreed this feature would be amazing

**Comment by DanielHo4** [comment]
agree, multiple monitor streaming is a fantastic feature

**Comment by Kofa1** [comment]
Agree.
Devs, please!

**Comment by roguebond** [comment]
Yup. This feature would be a total mic drop with a fat ass thud. We need this. Please.

**Comment by Pan-zc** [comment]
I strongly support the addition of multi-monitor support to Sunshine, as it would significantly enhance productivity for users working across multiple displays.

**Comment by ns6089** [comment]
It will require shifting project goals, currently sunshine is "Self-hosted game stream host for Moonlight."

For the more technical parts - developing new protocol (currently sunshine/moonlight operates on nvidia gamestream protocol with very minor extensions), updating sunshine and a dozen of its capture backends to that new protocol, updating moonlight library and updating each moonlight client (while maintaining existing moonlight compatibility with nvidia gamestream).

It may happen one day, but I don't see it happening anytime soon.

**Reply by utkvishwas** [reply]
Okayt i understand and thanksn for the work you guys do but what about touch driver support like SuperDisplay App on android?

**Comment by lalebarde** [comment]
Would be very very usefull

**Comment by Msprg** [comment]
> touch driver like SuperDisplay app

I for once, want this feature the most. Believe it or not, there are very valid use-cases for this - I could be running some games streamed from PC to my phone, but can't use the phone's touchscreen, despite the PC game supporting multitouch touchscreens.

**Reply by Dampfwalze** [reply]
I found that Moonlight interpreters touch events as mouse events, but stylus events are forwarded as actual stylus events.

There is a setting to switch the touch behavior from mouse pad like to klick where you pressed, but there seems to be no option to forward touch events as actual touch events to the Host.

I too would really appreciate the possible to switch that, because I use it on a Tablet with connected keyboard and mouse and stuff like scrolling would work with the finger too then.

**Comment by re0phimes** [comment]
agree, it has been a year since this issue posted, I really want to know if multi-screen support is in the schedule?

**Reply by ReenigneArcher** [reply]
Sunshine is open source... If you want something, do it

**Reply by Kofa1** [reply]
> Sunshine is open source... If you want something, do it

a.k.a "if you're homeless, just buy a house."

I've tried to learn coding and didn't like it at all. Would be cool if you'd create a language which everyone can understand without university education.

**Reply by gschintgen** [reply]
You're free to ask for a refund.

**Comment by stefanilijev97** [comment]
There is a workaround for that.. just create a virtual secondary monitor and specify it in the display options (tell Sunshine what screen to mirror) - also, that way you can also pick whatever resolution and refresh rate you want, no need for a hdmi dummy or something hardware related.

**Reply by Dampfwalze** [reply]
Well, that's a good option, but this is not a solution to the initial reqeust. It asked specifically for multiple monitor support, which you can't achieve just by some simple hack... Both sides have to support this.

But regarding virtual monitors, there are multiple open source drivers you could use to achieve that. One notable one is [virtual-display-rs](https://github.com/MolotovCherry/virtual-display-rs).

**Reply by Kofa1** [reply]
That requires running 2 instances of client and server side software at once. Not really a solution.

**Reply by Pheggas** [reply]
Is this valid for Mac host?

**Comment by devluz** [comment]
I am curious if just running one instance of Sunshine and Moonlight per monitor is really such a bad solution? We could have one main instance for handling input + main display and then a light version for additional screens.
Is there anything that would block two concurrent instances of Sunshine? (besides a change in ports)

**Reply by Msprg** [reply]
I am also using sunsh/moonlight as a remote desktop / access tool on my android phone. Try to manage one instance of moonlight per monitor on android.

Besides the mentioned multitouch support, even tho possibly unrelated matter, also matters to me. Whether it be 2 separate issues or only one, I still long for both features :(

**Reply by willow-sprite** [reply]
I can confirm there isn't anything stopping you from running 2 instances of sunshine concurrently for 2 separate monitors.
just have them pointed to different config files with different port numbers, it's even possible to play splitscreen games with nucleus coop.

you do need 1 of 3 things for a decent experience though.

1. a good modem, a base modem provided by your isp may not cut it, mine certainly doesn't, the 5G channel gets taken out by the traffic.
2. a decent network card + close proximity to the devices you're streaming to. use the pc as a hotspot. this is what I do.
3. everything connected to the modem through ethernet cables, this should work.

ah wait... 4th thing just occurred to me.
4. patience & troubleshooting skills.

**Comment by GPBeta** [comment]
Agree, I hope Sunshine would have any of the following features for multi-monitor support in the future:
- Allow selecting multiple monitors, and merging them into one video stream.
- Allow customizing capture area which may span across multiple screens.

**Comment by Zoewingz** [comment]
This is a very practical proposal because it is currently easy to achieve in Steam Link. My request is to be able to freely switch between multiple screens with the mouse movement, and to have the cursor move at a higher frame rate (the only drawback of Steam Link is that the mouse delay is too high now), or is there any way to solve the mouse delay problem of Steam Link, because I have the needs of multi-screen and remote digital painting.

**Reply by willow-sprite** [reply]
not sure if you're still here, but maybe use kde-connect to send low latency mouse inputs concurrently with steam link for multi-monitor, I'm not sure exactly how to split the inputs but it could work.

**Comment by iso2013** [comment]
Is there need for a new protocol if both monitors were appended together into one stream?

What I mean is that we could for example stream two side-by-side 1920x1080 monitors as one 3840x1080 monitor.

While this would mean your monitors have to be horizontally or vertically aligned, I don't think that's so much of a problem.

Yes, it might already be possible to do this by configuring the two displays as one through an external tool, but that would be relatively difficult compared to this solution—especially because it requires doing so on both the host and the client if there's no support on either end for splitting the stream across multiple monitors.


Another solution would be to use two separate streams, but have "one server". Not sure how much of a difference it would make over just having two copies of sunshine open in terms of performance, but the interface would be cleaner for users and I could see being able to adjust the bitrate by monitor being used quite a bit.


If either of these is a solution that would be considered I'm happy to take a look at implementing it, but being completely unfamiliar with the codebase it realistically would take me much longer than an existing contributor.

**Comment by reynadev-oss** [comment]
I believe that broadcasting over many monitors is a great function.
